pkg/models/registries/image.go

```diff
@@ -30,9 +30,45 @@ func (i *Image) Reference() string {
 	return i.Tag
 }
 
+// Maybe rawurl is of the form scheme:path.
+// (Scheme must be [a-zA-Z][a-zA-Z0-9+-.]*)
+// If so, return scheme, path; else return "", rawurl.
+func Getscheme(rawurl string) (scheme, path string, err error) {
```

Comment:
use https://golang.org/pkg/net/url/#URL ?

Reply:
`Getscheme`  is  a privately function in `net/url` package.
image name is not an URL, here just want to get scheme, and remove it

Reply:
I would prefer to use the standard library implementation, the code now looks complicated.
like:
```
type DockerURL struct {
	*url.URL
}

func (u *DockerURL) StringWithoutScheme() string {
	u.Scheme = ""
	s := u.String()
	return s[2:]
}

func ParseDockerURL(rawurl string) (*DockerURL, error) {
	url, _ := url.Parse(rawurl)
	return &DockerURL{URL: url}, nil
}
```

Reply:
this is a good way, but `rawurl` must contain "http://" or "https://". otherwise it will get a wrong result, such as `rawurl = nginx:latest`